### PR TITLE
Homework - Cars

### PR DESCRIPTION
**Pull request by DamiXioR**

### CODE REVIEW

**Review by MaciejSalamonski — APPROVED** [review]

**Review by WBigaj — APPROVED** [review]
I have checked your code and it looks good.

File: HybridCar-ut.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    EXPECT_NO_THROW(hybridCar.accelerate(444.9));
+}
+
+TEST_F(HybridCarTest, ShouldThrowInvalidAccelerateWhenTryToAccelerateWithNegativeNumber)
+{
+    EXPECT_THROW(hybridCar.accelerate(-500), InvalidAccelerate);
+    EXPECT_THROW(hybridCar.accelerate(-115.10), InvalidAccelerate);
+    EXPECT_THROW(hybridCar.accelerate(-54.0), InvalidAccelerate);
+    EXPECT_THROW(hybridCar.accelerate(-5.40), InvalidAccelerate);
+}
```
Comment (jakjan95) [inline HybridCar-ut.cpp:27]: Magic values!!

File: PetrolCar-ut.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+TEST_F(PetrolCarTest, ShouldThrowInvalidAccelerateWhenTryToAccelerateWithNegativeNumber)
+{
+    EXPECT_THROW(petrolCar.accelerate(-500), InvalidAccelerate);
+    EXPECT_THROW(petrolCar.accelerate(-115.10), InvalidAccelerate);
+    EXPECT_THROW(petrolCar2.accelerate(-54.0), InvalidAccelerate);
+    EXPECT_THROW(petrolCar2.accelerate(-5.40), InvalidAccelerate);
+}
+
```
Comment (jakjan95) [inline PetrolCar-ut.cpp:57]: same

**Review by jakjan95 — CHANGES_REQUESTED** [review]
Magic values needs fixes, also add description what you improved in this exercise

File: PetrolCar-ut.cpp (hunk 2 above)
Comment (DamiXioR) [reply PetrolCar-ut.cpp:57]: i think its not important in tests to set a lot of values variables, maybe i am wrong :)

File: HybridCar-ut.cpp (hunk 1 above)
Comment (DamiXioR) [reply HybridCar-ut.cpp:27]: yes, in tests

File: PetrolCar-ut.cpp (hunk 2 above)
Comment (ziobron) [reply PetrolCar-ut.cpp:57]: If you change it to parametrized test then you can have such magic values. Normally avoid it.

File: HybridCar.hpp [hunk 3]
```diff
@@ -1,19 +1,14 @@
+#pragma once
+
 #include "ElectricEngine.hpp"
 #include "PetrolEngine.hpp"
+#include "ElectricCar.hpp"
+#include "PetrolCar.hpp"
 
-class HybridCar
+class HybridCar : public virtual  ElectricCar, public virtual  PetrolCar
```
Comment (ziobron) [inline HybridCar.hpp:8]:
```suggestion
class HybridCar : public ElectricCar, public PetrolCar 
```
virtual is not needed here. Only ElectricCar and PetrolCar should inherit virtually.

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
4 XP za wyjątek InvalidGear w Cars (3521)
3 * 2 XP za poprawkę interfejsu w Cars (3522)
10 XP za testy w Cars (3523)
3 * 2 XP za punktualność (1119)
{ "requirement_ids": [3521, 3523, 3522, 3522, 3522, 1119, 1119, 1119] }

**Pull request closed without merging**